drivers/storage/portworx/component/disruption_budget.go

```diff
@@ -365,3 +375,122 @@ func RegisterDisruptionBudgetComponent() {
 func init() {
 	RegisterDisruptionBudgetComponent()
 }
+
+func (c *disruptionBudget) UpdateMinAvailableForNodePDB(cluster *corev1.StorageCluster,
+	ownerRef *metav1.OwnerReference,
+	nodeEnumerateResponse *api.SdkNodeEnumerateWithFiltersResponse,
+	k8sNodeList *v1.NodeList,
+) error {
+
+	// GetNodesToUpgrade returns list of nodes that can be upgraded in parallel and the map of px node to k8s node name, number of nodes with PDB 0 + down nodes
+	// In the case when non disruptive upgrade is disabled, it returns all cordoned nodes which aren't already selected for the upgrade
+	nodesToUpgrade, cordonedPxNodesMap, numPxNodesDown, err := pxutil.GetNodesToUpgrade(cluster, nodeEnumerateResponse, k8sNodeList, c.k8sClient, c.sdkConn)
+	if err != nil {
+		return err
+	}
+
+	// If user has mentioned minimum nodes that need to be running, then honor that value
+	storageNodesCount, err := pxutil.CountStorageNodes(cluster, c.sdkConn, c.k8sClient, nodeEnumerateResponse)
+	if err != nil {
+		c.closeSdkConn()
+		return err
+	}
+	userProvidedMinValue, err := pxutil.MinAvailableForStoragePDB(cluster)
+	if err != nil {
+		logrus.Warnf("Invalid value for annotation %s: %v", pxutil.AnnotationStoragePodDisruptionBudget, err)
+		userProvidedMinValue = -2
+	}
+
+	// Calculate the minimum number of nodes that should be available
+	quorumValue := int(math.Floor(float64(storageNodesCount)/2) + 1)
+	calculatedMinAvailable := quorumValue
+	// When non disruptive upgrades is disabled
+	if cluster.Annotations != nil && cluster.Annotations[pxutil.AnnotationsDisableNonDisruptiveUpgrade] == "true" {
+		if cluster.Annotations[pxutil.AnnotationStoragePodDisruptionBudget] == "" {
+			calculatedMinAvailable = storageNodesCount - 1
+		} else if (userProvidedMinValue < quorumValue || userProvidedMinValue >= storageNodesCount) && userProvidedMinValue != c.annotatedMinAvailable {
+			calculatedMinAvailable = storageNodesCount - 1
+			errmsg := fmt.Sprintf("Invalid minAvailable annotation value for storage pod disruption budget. Using default value: %d", calculatedMinAvailable)
+			c.recorder.Event(cluster, v1.EventTypeWarning, util.InvalidMinAvailable, errmsg)
+		} else if userProvidedMinValue >= quorumValue && userProvidedMinValue < storageNodesCount {
+			calculatedMinAvailable = userProvidedMinValue
+		}
+	} else {
+		// When non disruptive upgrades is enabled
+		if userProvidedMinValue >= storageNodesCount && userProvidedMinValue != c.annotatedMinAvailable {
```

Comment:
What i felt was, when non disruptive upgrades is disabled, and minAvailable is set, all of those nodes will be selected for upgrade directly. So it is best to let the user know their lower value is invalid
But in the case of non disruptive upgrades, we only upgrade as many nodes as possible without disruption, so px quorum value or lower doesn't really matter here . This might not really concern the user much as they know its a best efforts upgrade.